queries.sql

```diff
@@ -1,3 +1,153 @@
+LEFT JOIN animals
+  ON species.id = animals.species_id
+WHERE animals.species_id = 1;
+
+SELECT full_name, name FROM owners o
+LEFT JOIN animals a
+  ON o.id = a.owner_id
+ORDER BY full_name;
+
+SELECT species.name, COUNT(animals.name) FROM species
+LEFT JOIN animals
+  ON species.id = animals.species_id
+GROUP BY species.name;
+
+SELECT o.full_name, s.name, a.name FROM animals a
+INNER JOIN owners o
+  ON o.id = a.owner_id
+INNER JOIN species s
+  ON s.id = a.species_id
+WHERE owner_id = 2 AND species_id = 2;
+
+SELECT full_name, name FROM owners o
+LEFT JOIN animals a
+  ON o.id = a.owner_id
+WHERE owner_id = 5 AND escape_attempts = 0;
+
+SELECT full_name, COUNT(name) FROM owners o
+LEFT JOIN animals a
+  ON a.owner_id = o.id
+GROUP BY full_name;
+
+-- QUESTIONS RELATED TO VISITS AND VETS - DAY 4
+
+SELECT v.name as vet, a.name as animal, vis.date_of_visit FROM visits vis
+INNER JOIN animals a
+  ON a.id = vis.animals_id
+INNER JOIN vets v
+  ON v.id = vis.vets_id
+WHERE vis.date_of_visit = (select max(date_of_visit) FROM visits WHERE vets_id=1) 
+ORDER BY vis.date_of_visit desc;
+
+SELECT count(a.name) FROM visits vis
+INNER JOIN animals a
+  ON a.id = vis.animals_id
+INNER JOIN vets v
+  ON v.id = vis.vets_id
+WHERE v.name = 'Stephanie Mendez';
+
+SELECT v.name, s.name FROM specializations sp
+INNER JOIN species s
+  ON s.id = sp.species_id
+RIGHT JOIN vets v
+  ON v.id = sp.vets_id;
+
+SELECT v.name, a.name, vis.date_of_visit FROM visits vis
+INNER JOIN animals a
+  ON a.id = vis.animals_id
+INNER JOIN vets v
+  ON v.id = vis.vets_id
+WHERE v.name = 'Stephanie Mendez' and date_of_visit BETWEEN 'April 1, 2020' and 'August 30, 2020';
+
+SELECT a.name, count(a.name) FROM visits vis
+INNER JOIN animals a
+  ON a.id = vis.animals_id
+INNER JOIN vets v
+  ON v.id = vis.vets_id
+GROUP BY a.name
+ORDER BY count desc
+LIMIT 1;
+
+SELECT v.name, a.name, date_of_visit FROM visits vis
+INNER JOIN animals a
+  ON a.id = vis.animals_id
+INNER JOIN vets v
+  ON v.id = vis.vets_id
+WHERE date_of_visit = (select min(date_of_visit) FROM visits WHERE vets_id = 2)
+ORDER BY date_of_visit;
+
+SELECT a.name, v.name, date_of_visit  FROM visits vis
+INNER JOIN animals a
+  ON a.id = vis.animals_id
+INNER JOIN vets v
+  ON v.id = vis.vets_id
+WHERE date_of_visit = (select max(date_of_visit) FROM visits);
+
+
+SELECT v.name, count(a.name) FROM visits vis
+INNER JOIN animals a
+  ON a.id = vis.animals_id
+INNER JOIN vets v
+  ON v.id = vis.vets_id
+WHERE v.id not in (select vets_id FROM specializations)
+GROUP BY v.name;
+
+
+SELECT v.name, s.name, count(a.species_id) FROM visits vis
+INNER JOIN animals a
+  ON a.id = vis.animals_id
+INNER JOIN vets v
+  ON v.id = vis.vets_id
+INNER JOIN species s
+  ON s.id = a.species_id
+WHERE v.name = 'Maisy Smith'
+GROUP BY (v.name, s.name)
+ORDER BY count desc
+LIMIT 1;
+
+--week 2
+-- query plans 
+
+EXPLAIN ANALYZE SELECT COUNT(*) FROM visits where animals_id = 4;
+EXPLAIN ANALYZE SELECT * FROM visits where vets_id = 2;
```

Comment:
@BenMukebo and there is not in requirement that after doing performance ms need to be lower than 1000. And as you can see I imported a lot more raws (in every 1 hour added dates of visits starting from the 1900-01-01') that was needed more than 8 000 000, so it is normal than after performance it still need a lot of time. I added more raws than needed because my code first was running very fast, faster than 1000ms, in requirement was that it need to be more that 1000 before the performance. I appreciate your review, but please take a better look at what I did. You wrote that I didn't add a screenshot after the performance, but I did (I don't know regarding of what you got it). And, there is only 4 vets, which means that after the performance for this search it will be not much improved when there is a lot of row like in my case.

Reply:
@BenMukebo and there is not in requirement that after doing performance ms need to be lower than 1000. And as you can see I imported a lot more raws (in every 1 hour added dates of visits starting from the 1900-01-01') that was needed more than 8 000 000, so it is normal than after performance it still need a lot of time. I added more raws than needed because my code first was running very fast, faster than 1000ms, in requirement was that it need to be more that 1000 before the performance. I appreciate your review, but please take a better look at what I did.